krun/scheduler.py

```diff
@@ -360,41 +386,39 @@ def __init__(self, config, mailer, platform, dry_run=False,
         self.log_path = self.config.log_filename(not self.on_first_invocation)
 
         if on_first_invocation:
-            self.manifest = ManifestManager(config, new_file=True)
-            self.results = Results(self.config, self.platform)
+            self.manifest = ManifestManager(config, platform, new_file=True)
         else:
-            self.manifest = ManifestManager(self.config)
-            self.results = None
+            self.manifest = ManifestManager(self.config, platform)
 
-    def get_estimated_exec_duration(self, key):
-        previous_exec_times = self.results.eta_estimates.get(key)
+    def get_estimated_exec_duration(self, results, key):
```

Comment:
The ETA info is in the results file. This was stored in self, but I removed it so we could more easily see the extent of its use.